src/main/java/com/gradle/CustomBuildScanEnhancements.java

```diff
@@ -248,7 +248,7 @@ private void execute() {
                     addCustomValueAndSearchLink(buildScan, "CI workflow", value));
                 envVariable("GITHUB_RUN_ID").ifPresent(value ->
                         addCustomValueAndSearchLink(buildScan, "CI run", value));
-                envVariable("GITHUB_HEAD_REF").ifPresent(value ->
+                envVariable("GITHUB_HEAD_REF").filter(value -> !value.isEmpty()).ifPresent(value ->
```

Comment:
can we do the filter after the `ifPresent` check? since we wouldn't even need the filter if the value isn't present

Reply:
Sure, however, filter works on the `Option`, so if that's null it won't do anything.

Reply:
oh i see, in that case does that mean the `ifPresent` isn't needed?

Reply:
It still is, the filter will return `empty` if the value doesn't pass the check.